app/controllers/application_controller.rb

```diff
@@ -6,6 +6,27 @@ class ApplicationController < ActionController::Base
 
   private
 
+  def logged_in_user
+    return if logged_in?
+    store_location
+    flash[:danger] = t("pli")
+    redirect_to login_url
+  end
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password,
+      :password_confirmation)
+  end
+
+  def correct_user
+    @user = User.find_by(params[:id])
```

Comment:
em sửa lại như này nhé:
```suggestion
    @user = User.find_by id: params[:id]
```

Reply:
dạ a